### PR TITLE
Improve performance of metatag doesnt exist expr

### LINKED ISSUE #1871: Tag queries with "metatag does not exist" style expression extremely slow
Tag queries with an expression of the form `metatag=` (i.e. `metatag` does not exist) incredibly slow. Caused index lock ups. the example in this case was `[namespace^=b stage=]` where `stage` is the meta tag. We have about 50k meta tag records with `stage` in it.

### PR DESCRIPTION
**Pull request by replay**
Fixes #1871 

This PR's branch is based on top of the branch of PR #1898, it only makes sense to review this PR once #1898 is merged.

This adds an optimization which is specific for queries that query for the absence of a meta tag of which all underlying expressions are using the `=` operator. This optimization currently already exists for queries which query for the presence of a meta tag using `=` and of which all underlying expressions are also using the `=` operator, in this PR that existing optimization gets extended to also work for filter expressions which check for the absence of a meta tag.

I also added a benchmark to test for that specific case, and It shows significant improvements with/without this change:

```
benchmark                                old ns/op        new ns/op     delta
BenchmarkFilterByAbsenceOfMetaTag-12     207231916200     253464875     -99.88%

benchmark                                old allocs     new allocs     delta
BenchmarkFilterByAbsenceOfMetaTag-12     407059         232777         -42.81%

benchmark                                old bytes     new bytes     delta
BenchmarkFilterByAbsenceOfMetaTag-12     37499976      34272224      -8.61%
```

### CODE REVIEW

**Comment by Dieterbe** [comment]
ping @shanson7

**Comment by shanson7** [comment]
The changes look good to me and the improvement numbers look fantastic.

**Comment by Dieterbe** [comment]
this PR needs an approval before it can be merged.
looks like @shanson7 has basically reviewed it already, and i suspect may be running this in prod already, so sean if you could approve this PR.. thanks :)

**Comment by shanson7** [comment]
We have not rolled these changes out yet. We have ben blocked on another item for a little while. I plan on doing a rollout with this and a few other recent improvements soon.

**Review by shanson7 — APPROVED** [review]
LGTM

**Pull request merged**
